### ISSUE TITLE
Under python3, Frescobaldi can't open files with non-ascii characters

### ISSUE THREAD

**Issue by wbsoft** [wbsoft/frescobaldi#562]
Catched by proski:
Frescobaldi under python 3 cannot open files with umlauts. QApplication is instantiated with correct sys.argv in app.py, but QApplication.arguments() in main.py returns arguments with umlauts stripped away. I don't see any locale settings for QApplication. Locale is en_US.UTF-8. Observed on Ubuntu 14.10 and Fedora 21.
Mentioned in a comment below pull request #557.

**Comment by proski** [comment]
I have found a fix, but it's quite ugly. For whatever reason, QApplication cannot be initialized with arguments as unicode strings. So sys.argv should be translated to bytes on Python 3 as suggested in http://bugs.python.org/issue8776

I'd rather persuade `QApplication.__init__` to take unicode strings. I haven't found anything relevant in the Python sources of python3-qt4, I guess it's in the C sources. Maybe sip.setapi() magic is needed.

```
Fix opening files with non-ascii names in Python 3

From: Pavel Roskin <proski@gnu.org>

Instantiate QApplication with bytes, it cannot take unicode
---
 frescobaldi_app/app.py |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/frescobaldi_app/app.py b/frescobaldi_app/app.py
index 121040f..b1c291d 100644
--- a/frescobaldi_app/app.py
+++ b/frescobaldi_app/app.py
@@ -98,7 +98,12 @@ def findDocument(url):
 def instantiate():
     """Instantiate the global QApplication object."""
     global qApp
-    qApp = QApplication([os.path.abspath(sys.argv[0])] + sys.argv[1:])
+
+    if sys.version_info[0] > 2:
+        argvb = list(map(os.fsencode, sys.argv))
+    else:
+        argvb = sys.argv
+    qApp = QApplication([os.path.abspath(argvb[0])] + argvb[1:])
     QApplication.setApplicationName(info.name)
     QApplication.setApplicationVersion(info.version)
     QApplication.setOrganizationName(info.name)
```

**Comment by wbsoft** [comment]
I was also thinking this route, but I will make args[0] absolute before encoding to binary.

**Comment by tobik** [comment]
Should this be fixed in Frescobaldi 2.18?

**Comment by proski** [comment]
Yes, it should be fixed. If it doesn't work for you, please open a separate issue.
